scientific-computing-with-python/learn-classes-and-objects-by-building-a-sudoku-solver/sudoku_solver.py

```diff
@@ -0,0 +1,122 @@
+class Board:
+    def __init__(self, board):
+        self.board = board
+        self.print_board()
+    
+    # Find the first empty square (currently containing zero)
+    def find_empty_square(self):
+        for row, lst in enumerate(self.board):
+            if 0 in lst:
+                col = lst.index(0)
+                return row, col

```

Comment:
```suggestion
        for row, contents in enumerate(self.board):
            if 0 in contents:
                col = contents.index(0)
                return row, col
```
or maybe, 
```suggestion
        for row_no, row in enumerate(self.board):
            if 0 in row:
                col_no = row.index(0)
                return row_no, col_no

```

---
I'd consider also using `try/except` here. As a part of the _Easier to ask for forgiveness than permission_ Python convention.
```suggestion
        for row_no, row in enumerate(self.board):
            try:
                col = row.index(0)
                return row_no, col
            except ValueError:
                pass
```